### DISCUSSION TITLE
run recorded ironpython script through pyaedt

### DISCUSSION THREAD

**Discussion by AghHossein** [ansys/pyaedt#3025] [Q&A] [answered]
Is it possible to run a script that is already wrirren in ironpython for aedt through pyaedt?

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @AghHossein ,

Yes it is possible. You just need to change the equivalent pyaedt object for example:

app = Hfss()

oDesign = app.odesign

And the variable oDesign will be equivalent to the Ironpython one. Please if you do dir(app) you will see all Ironpython objects inherited.

Please close the comment if you think I solved it. 

Maybe this picture helps you (I created it 2 years ago, and maybe has changed something)

<img width="1438" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/4538037d-5072-46aa-bfaa-6473ace9ebb2">

**Reply by shabham22** [reply]
Hi @Samuelopez-ansys,

Can you please provide a short example in which both PyAEDT and Ironpython is used in single script. (if this type of any example is already there in Ansys help section, kindly provide link)

Like i want to open/load an .aedt file using PyAEDT and in that i want to run some recorded operation like ChangeProperty using Ironpython with the same program. Can I run using `import subprocess`

Also, kindly let me know for ironpython do I need to install something? (like for PyAED, we install `pip install pyaedt`)

**Reply by Samuelopez-ansys** [reply]
Hi @shabham22,

Ironpython is the native Interpreter in AEDT, so you do not need to install anything else. But it has some limitations, for instance you can not debug de code in external IDE such as Visual Studio or PyCharm. I agree the recording script is convenient when you do not know how to start, but I encourage to use PyAEDT syntax if it is available.

When you say "Ironpython" commands, you mean Native AEDT API. The Native AEDT API is compatible with Ironpython and CPython. CPython is the Python interpreter that all open source libraries are using, and it is the main reason of using PyAEDT, because Ironpython does not allow you to use open source libraries such as matplotlib, numpy...

PyAEDT main goal is wrapping Native AEDT API, and expose it to the user in a much simpler way, then your code will be much cleaner and much easier to maintain.

For instance, you can use the following PyAEDT code to open an aedt file, it assumes you have a project opened:

```
import ansys.aedt.core
from ansys.aedt.core import get_pyaedt_app

# PyAEDT syntax to connect to AEDT
app = ansys.aedt.core.Desktop(version="2024.2")

active_project = app.active_project()
active_design = app.active_design()

# Get active project and active design using Native AEDT API:
project_name = active_project.GetName()
if active_design.GetDesignType() == "HFSS 3D Layout Design":
    design_name = active_design.GetDesignName()
else:
    design_name = active_design.GetName()

# Get PyAEDT object (it could be HFSS, Icepak...)
aedtapp = get_pyaedt_app(project_name, design_name, app)

# Release desktop once you finish
app.release_desktop()
```

In the User guide you can find multiple combinations:
https://aedt.docs.pyansys.com/version/stable/User_guide/intro.html

And also in the examples:
https://examples.aedt.docs.pyansys.com/
